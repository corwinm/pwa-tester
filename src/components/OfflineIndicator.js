import React from "react";
import { useOffline } from "../custom-hooks/useOffline";

export default function OfflineIndicator() {
  const offline = useOffline();

  if (!offline) {
    return null;
  }
  return (
    <section className="notch-inset-item has-background-grey-lighter ">
      <h1 className="navbar-item has-text-danger">Offline</h1>
    </section>
  );
}
