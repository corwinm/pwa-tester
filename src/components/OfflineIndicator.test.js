import React from "react";
import { render, cleanup } from "react-testing-library";
import OfflineIndicator from "./OfflineIndicator";

afterEach(cleanup);

jest.mock("../custom-hooks/useOffline");
const useOfflineMock = require("../custom-hooks/useOffline");

test("matches online snapshot", () => {
  useOfflineMock.useOffline.mockImplementation(() => false);
  const { container } = render(<OfflineIndicator />);
  expect(container.firstChild).toMatchSnapshot();
});

test("matches offline snapshot", () => {
  useOfflineMock.useOffline.mockImplementation(() => true);
  const { container } = render(<OfflineIndicator />);
  expect(container.firstChild).toMatchSnapshot();
});