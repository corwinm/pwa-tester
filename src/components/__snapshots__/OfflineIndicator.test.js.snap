// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches offline snapshot 1`] = `
<section
  class="notch-inset-item has-background-grey-lighter "
>
  <h1
    class="navbar-item has-text-danger"
  >
    Offline
  </h1>
</section>
`;

exports[`matches online snapshot 1`] = `null`;
